Arm64 builds failing in Docker for v0.4.0

I can't get v0.4.0 to successfully build for arm64 in Docker due to this error:

https://github.com/sethforprivacy/phoenixd-docker/actions/runs/11326427746/job/31495346699#step:10:442

https://paste.sethforprivacy.com/?09fb59720819f937#AxdwwoXQZ5s8VciBf942cJVRFbisvKPRTGptLQTMYXE9

Not quite sure where to get started fixing it, but should be easy to reproduce as our Dockerfiles are practically identical.

Tested that, failed. Here are the logs:


```
#13 25.60 Starting a Gradle Daemon (subsequent builds will be faster)
#13 304.8 > Task :buildSrc:checkKotlinGradlePluginConfigurationErrors
#13 334.9 > Task :buildSrc:compileKotlin
#13 356.8 > Task :buildSrc:pluginDescriptors
#13 356.9 > Task :buildSrc:processResources NO-SOURCE
#13 399.6 > Task :buildSrc:compileJava NO-SOURCE
#13 399.6 > Task :buildSrc:compileGroovy NO-SOURCE
#13 399.6 > Task :buildSrc:classes UP-TO-DATE
#13 400.0 
#13 400.0 > Task :buildSrc:jar
#13 400.0 :jar: No valid plugin descriptors were found in META-INF/gradle-plugins
#13 519.3 
#13 519.3 > Configure project :
#13 519.3 architecture=aarch64
#13 521.6 SQLDelight Gradle plugin was applied but there are no databases set up.
#13 523.4 
#13 523.4 FAILURE: Build failed with an exception.
#13 523.4 
#13 523.4 * Where:
#13 523.4 Build file '/phoenixd/build.gradle.kts' line: 194
#13 523.4 
#13 523.4 * What went wrong:
#13 523.4 Task with name 'linuxX64Binaries' not found in root project 'phoenixd'.
#13 523.4 
#13 523.4 * Try:
#13 523.4 > Run with --stacktrace option to get the stack trace.
#13 523.4 > Run with --info or --debug option to get more log output.
#13 523.4 > Run with --scan to get full insights.
#13 523.4 > Get more help at https://help.gradle.org./
#13 523.4 
#13 523.4 BUILD FAILED in 8m 41s
#13 523.4 4 actionable tasks: 4 executed
#13 ERROR: process "/bin/sh -c git clone --recursive --single-branch --branch ${PHOENIXD_BRANCH} -c advice.detachedHead=false     https://github.com/ACINQ/phoenixd .     && test `git rev-parse HEAD` = ${PHOENIXD_COMMIT_HASH} || exit 1     && ./gradlew distTar" did not complete successfully: exit code: 1
------
 > [build 5/5] RUN git clone --recursive --single-branch --branch linux-arm-build-0.4.0 -c advice.detachedHead=false     https://github.com/ACINQ/phoenixd .     && test `git rev-parse HEAD` = b5cd95c54dd2d2073fbfd7945f98334077e12a57 || exit 1     && ./gradlew distTar:
523.4 Task with name 'linuxX64Binaries' not found in root project 'phoenixd'.
523.4 
523.4 * Try:
523.4 > Run with --stacktrace option to get the stack trace.
523.4 > Run with --info or --debug option to get more log output.
523.4 > Run with --scan to get full insights.
523.4 > Get more help at https://help.gradle.org./
523.4 
523.4 BUILD FAILED in 8m 41s
523.4 4 actionable tasks: 4 executed
------
Dockerfile:17
--------------------
  16 |     WORKDIR /phoenixd
  17 | >>> RUN git clone --recursive --single-branch --branch ${PHOENIXD_BRANCH} -c advice.detachedHead=false \
  18 | >>>     https://github.com/ACINQ/phoenixd . \
  19 | >>>     && test `git rev-parse HEAD` = ${PHOENIXD_COMMIT_HASH} || exit 1 \
  20 | >>>     && ./gradlew distTar
  21 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c git clone --recursive --single-branch --branch ${PHOENIXD_BRANCH} -c advice.detachedHead=false     https://github.com/ACINQ/phoenixd .     && test `git rev-parse HEAD` = ${PHOENIXD_COMMIT_HASH} || exit 1     && ./gradlew distTar" did not complete successfully: exit code: 1
Error: buildx failed with: ERROR: failed to solve: process "/bin/sh -c git clone --recursive --single-branch --branch ${PHOENIXD_BRANCH} -c advice.detachedHead=false     https://github.com/ACINQ/phoenixd .     && test `git rev-parse HEAD` = ${PHOENIXD_COMMIT_HASH} || exit 1     && ./gradlew distTar" did not complete successfully: exit code: 1
```

--- a/README.md
+++ b/README.md
@@ -10,7 +10,7 @@ It is written in [Kotlin Multiplatform](https://kotlinlang.org/docs/multiplatfor
 
 ## Build
 
-### Native Linux/WSL
+### Native Linux/WSL x64
 
 Requires `libsqlite-dev` and `libcurl4-gnutls-dev`, both compiled against `glibc 2.19`.
 

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -71,7 +71,6 @@ val buildVersionsTask by tasks.registering(Sync::class) {
 
 val currentOs = org.gradle.internal.os.OperatingSystem.current()
 val arch = System.getProperty("os.arch")
-println("architecture=$arch")
 
 kotlin {
     jvm {
@@ -202,7 +201,7 @@ distributions {
             }
         }
     }
-    if (currentOs.isLinux) {
+    if (currentOs.isLinux && arch != "aarch64") {
         create("linuxX64") {
             configureNativeDistribution("linuxX64Binaries", "linuxX64", "linux-x64")
         }

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -71,6 +71,7 @@ val buildVersionsTask by tasks.registering(Sync::class) {
 
 val currentOs = org.gradle.internal.os.OperatingSystem.current()
 val arch = System.getProperty("os.arch")
+println("architecture=$arch")
 
 kotlin {
     jvm {

